panther_lights/test/test_driver_node.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <image_transport/image_transport.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/image_encodings.hpp>
+#include <thread>
+
+#include <panther_lights/test/mock_driver_node.hpp>
+#include <panther_msgs/srv/set_led_brightness.hpp>
+#include <panther_utils/test/ros_test_utils.hpp>
+
+using ImageMsg = sensor_msgs::msg::Image;
+using SetLEDBrightnessSrv = panther_msgs::srv::SetLEDBrightness;
+using MockDriverNode = panther_lights::mock_driver_node::MockDriverNode;
+
+class DriverNodeFixture : public ::testing::Test
+{
+public:
+  // Due to DriverNode::OnShutdown() we need to execute rclcpp::shutdown()
+  // in TearDown() to release GPIO pins
+  void SetUp() override
+  {
+    rclcpp::init(0, nullptr);
+    node_ = std::make_shared<MockDriverNode>("test_lights_driver_node");
+    it_ = std::make_shared<image_transport::ImageTransport>(node_->shared_from_this());
+    channel_1_pub_ = std::make_shared<image_transport::Publisher>(
+      it_->advertise("lights/driver/channel_1_frame", 5));
+    channel_2_pub_ = std::make_shared<image_transport::Publisher>(
+      it_->advertise("lights/driver/channel_2_frame", 5));
+    set_brightness_client_ =
+      node_->create_client<SetLEDBrightnessSrv>("lights/driver/set/brightness");
+    node_->Initialize();
+  }
+
+  void TearDown() override { rclcpp::shutdown(); }
```

Comment:
Is there a reason to use SetUp and TearDown instead of constructor and deconstructor?